pkg/build/generator/generator.go

```diff
@@ -432,6 +432,14 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	// Retry getting a non-conflicting name 10 times
```

Comment:
Can you add an explanation here of what happens when trying the next 10 numbers fails?